### PR TITLE
redecode events & nightly windows bugfix

### PR DESCRIPTION
**Pull request by jbrit**
Should close the issues:
-  [Redecode Events](https://github.com/orgs/rotki/projects/11/views/2?pane=issue&itemId=43187211) 
- [Windows Nightly](https://github.com/orgs/rotki/projects/11/views/2?pane=issue&itemId=43187326)

### CODE REVIEW

**Comment by jbrit** [comment]
sqldiff
```sql
INSERT INTO evm_events_info(identifier,tx_hash,counterparty,product,address,extra_data) VALUES(7352,'hash',NULL,NULL,'0',NULL);
INSERT INTO evm_events_info(identifier,tx_hash,counterparty,product,address,extra_data) VALUES(7353,'hash',NULL,NULL,'0',NULL);
INSERT INTO evm_transactions(identifier,tx_hash,chain_id,timestamp,block_number,from_address,to_address,value,gas,gas_price,gas_used,input_data,nonce) VALUES(1,'hash',0,1658912400020,0,'0x0000000000000000000000000000000000000000','0x0000000000000000000000000000000000000000','0.889','98','987','9','',0);
INSERT INTO history_events(identifier,entry_type,event_identifier,sequence_index,timestamp,location,location_label,asset,amount,usd_value,notes,type,subtype) VALUES(7352,1,'RE_TEST_CUSTOM',0,1658912400011,'A',NULL,'ETH','1','0','Test Custom deposit','deposit','spend');
INSERT INTO history_events(identifier,entry_type,event_identifier,sequence_index,timestamp,location,location_label,asset,amount,usd_value,notes,type,subtype) VALUES(7353,1,'RE_TEST',0,1658912400020,'A',NULL,'ETH','1','0','Test deposit','deposit','spend');
INSERT INTO history_events_mappings(rowid,parent_identifier,name,value) VALUES(1,7352,'state',1);
INSERT INTO history_events_mappings(rowid,parent_identifier,name,value) VALUES(2,7353,'state',0);
```

File: rotkehlchen/db/upgrades/v39_v40.py [hunk 1]
```diff
@@ -334,6 +339,38 @@ def _add_new_tables(write_cursor: 'DBCursor') -> None:
     log.debug('Exit _add_new_tables')
 
 
+def _reset_decoded_events(write_cursor: 'DBCursor') -> None:
+    """
+    Reset all decoded evm events except the customized ones for ethereum mainnet,
+    optimism and polygon.
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v39_v40.py:345]: Isn't it also for arbitrum by this release? This is copy pasta from v38->v39

File: rotkehlchen/db/upgrades/v39_v40.py [hunk 2]
```diff
@@ -334,6 +339,38 @@ def _add_new_tables(write_cursor: 'DBCursor') -> None:
...
+        'SELECT COUNT(*) FROM history_events_mappings WHERE name=? AND value=?',
+        (HISTORY_MAPPING_KEY_STATE, HISTORY_MAPPING_STATE_CUSTOMIZED),
+    ).fetchone()[0]
+    querystr = (
+        'DELETE FROM history_events WHERE identifier IN ('
+        'SELECT H.identifier from history_events H INNER JOIN evm_events_info E '
+        'ON H.identifier=E.identifier AND E.tx_hash IN '
+        '(SELECT tx_hash from_evm_transactions))'
+    )
+    bindings: tuple = ()
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v39_v40.py:361]: nitpick. I don't think type is needed here. It's infered

```suggestion
    bindings = ()
```

File: rotkehlchen/tests/api/blockchain/test_substrate.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ def test_add_ksm_blockchain_account_invalid(rotkehlchen_api_server):
 @pytest.mark.vcr()
 @pytest.mark.parametrize('number_of_eth_accounts', [0])
 @pytest.mark.parametrize('kusama_manager_connect_at_start', [(KusamaNodeName.OWN,)])
-@pytest.mark.parametrize('ksm_rpc_endpoint', [KUSAMA_TEST_RPC_ENDPOINT])
+@pytest.mark.parametrize('ksm_rpc_endpoint', [KUSAMA_TEST_RPC_ENDPOINT], ids=['KUSAMA_TEST_RPC_ENDPOINT'])  # setting ids to rename the argument to be processed by vcr # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/blockchain/test_substrate.py:54]: 1.This should have been split in 2 different commits. I said that during the standup. Why was it not done?
2. What does this do? I don't understand. How does this solve the problem? This is not explained at all here or in the other comments. Also don't just copy past the comment. Just explain that this affects all similar fixtures in the test file

**Review by LefterisJP — COMMENTED** [review]
some comments

File: rotkehlchen/db/upgrades/v39_v40.py (hunk 2 above)
Comment (jbrit) [reply rotkehlchen/db/upgrades/v39_v40.py:361]: lint fails if it isn't there

File: rotkehlchen/tests/api/blockchain/test_substrate.py (hunk 3 above)
Comment (jbrit) [reply rotkehlchen/tests/api/blockchain/test_substrate.py:54]: how it solves the problem, it changes the cassette name, I added more context here:
https://discord.com/channels/657906918408585217/1080783409640648765/1169173684305539134

File: rotkehlchen/db/upgrades/v39_v40.py (hunk 1 above)
Comment (jbrit) [reply rotkehlchen/db/upgrades/v39_v40.py:345]: yes

File: rotkehlchen/tests/api/blockchain/test_substrate.py (hunk 3 above)
Comment (jbrit) [reply rotkehlchen/tests/api/blockchain/test_substrate.py]: it's in split commits now

File: rotkehlchen/tests/api/blockchain/test_substrate.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ def test_add_ksm_blockchain_account_invalid(rotkehlchen_api_server):
 @pytest.mark.vcr()
 @pytest.mark.parametrize('number_of_eth_accounts', [0])
 @pytest.mark.parametrize('kusama_manager_connect_at_start', [(KusamaNodeName.OWN,)])
-@pytest.mark.parametrize('ksm_rpc_endpoint', [KUSAMA_TEST_RPC_ENDPOINT])
+@pytest.mark.parametrize('ksm_rpc_endpoint', [KUSAMA_TEST_RPC_ENDPOINT], ids=['KUSAMA_TEST_RPC_ENDPOINT'])  # setting ids to rename the argument to be processed by vcr, affects all other similar fixtures in this file # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/blockchain/test_substrate.py:54]:
```suggestion
@pytest.mark.parametrize('ksm_rpc_endpoint', [KUSAMA_TEST_RPC_ENDPOINT], ids=['KUSAMA_TEST_RPC_ENDPOINT'])  # setting ids to rename the argument to be processed by vcr since its value can contain characters that are illegal in windows. Affects all other similar fixtures in this file # noqa: E501
```

**Review by LefterisJP — APPROVED** [review]
Tiny nitpick. But can be done by @yabirgb in next PR. Single failed test is a timeout

File: rotkehlchen/tests/api/blockchain/test_substrate.py (hunk 4 above)
Comment (LefterisJP) [reply rotkehlchen/tests/api/blockchain/test_substrate.py:54]: I just added the change here: https://github.com/rotki/rotki/pull/6882
